reuse.go

```diff
@@ -3,27 +3,56 @@ package libp2pquic
 import (
 	"net"
 	"sync"
-	"sync/atomic"
+	"time"
 
 	"github.com/vishvananda/netlink"
 )
 
+// Constants. Defined as variables to simplify testing.
+var (
+	garbageCollectInterval = 30 * time.Second
+	maxUnusedDuration      = 10 * time.Second
+)
+
 type reuseConn struct {
 	net.PacketConn
-	refCount int32 // to be used as an atomic
+
+	mutex       sync.Mutex
+	refCount    int
+	unusedSince time.Time
 }
 
 func newReuseConn(conn net.PacketConn) *reuseConn {
 	return &reuseConn{PacketConn: conn}
 }
 
-func (c *reuseConn) IncreaseCount() { atomic.AddInt32(&c.refCount, 1) }
-func (c *reuseConn) DecreaseCount() { atomic.AddInt32(&c.refCount, -1) }
-func (c *reuseConn) GetCount() int  { return int(atomic.LoadInt32(&c.refCount)) }
+func (c *reuseConn) IncreaseCount() {
```

Comment:
nit: should probably be private.

Reply:
I don't think it actually matters. We're never exporting a `reuseConn` from this package, so there's no way a user could call this function anyway.

Reply:
Hm. But can't the user type assert?

Reply:
Not really. What we're returning is (wrapped) QUIC connections, not UDP connections.

Reply:
Ah, fair enough. I still think private functions are clearer but, well, we're not consistent about that anyways.